pkg/softimpersonation/softimpersonation.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package softimpersonation
```

Comment:
Can we get a `doc.go` to explain what this package does and why it's needed?

Reply:
Isn't the doc on the file itself sufficient, since there's only 1 file ?
See the `metadata` package for example.